app/src/main/java/com/onegini/mobile/exampleapp/view/activity/AuthenticationActivity.java

```diff
@@ -115,25 +116,36 @@ private boolean isBlank(final String string) {
 
   @Override
   public void onBackPressed() {
-    if(isCancellable()){
+    if (isCancellable()) {
       cancelRequest();
       super.onBackPressed();
     }
   }
 
-  protected boolean isCancellable(){
+  protected boolean isCancellable() {
     //by default we don't want to be able to go back from the pin screen,
     //but if cancel button is visible, we want to be able to use back button as well
     return isCancelButtonVisible();
   }
 
-  private boolean isCancelButtonVisible(){
-    if(cancelButton == null || cancelButton.getVisibility() == View.GONE){
+  private boolean isCancelButtonVisible() {
+    if (cancelButton == null || cancelButton.getVisibility() == View.GONE) {
       return false;
     }
     return true;
   }
 
-  protected void cancelRequest(){
+  protected void setCancelButtonVisibility(final int visibility) {
```

Comment:
If we pass `VISIBLE`/`GONE` only (no `INVISIBLE`) than maybe it's easier to pass a boolean. Then we don't need to validate if passed value is correct.